src/popup/modules/UserInterface.js

```diff
@@ -435,6 +437,37 @@ function triggerFileSave(file, filename, requestDownloadPermissions) {
     });
 }
 
+/**
+ * If qrcode text was an URL i.e contains :// this function generates a filename using the domain and replacing all special characters with _
+ *.If not return "qrcode"
+ *
+ * @function
+ * @private
+ * @param { } 
+ * @returns {filename}
+ */
+function generateFilename(){
+	var firstIndex = qrCodeInputText.indexOf("://");
+	if(firstIndex > 0) {
```

Comment:
Also per our contributing guide: use "early return" if styles, i.e. check for the inverse result and return the fallback.

Reply:
Sorry I learned that early return is bad style, but this was a few years ago.
But of course I can change this if you prefer.

Reply:
I intended it as the "error" return.

Reply:
Well… it's stated in the contribution guide, so please follow what is written there. That does, of course, not mean these rules are written into stone and may never be changed, but if we'd want to change this, that would be a thing we need to discuss first.
As for this case, I think – as usual – it is a matter of taste and possibly depends on other factors like the used programming language etc. As for this project, the “official taste” is to use early return.  The [Stackexchange discussion](https://softwareengineering.stackexchange.com/q/18454/209137) for this feature also lists some points on why this is useful, if you want to see some reasoning for that rule.

Reply:
So you want to have 2 error exits with the same return values?